src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -319,6 +319,24 @@ private TypeSymbol DecodeTypeOrThrow(ref BlobReader ppSig, SignatureTypeCode typ
                     typeSymbol = DecodeGenericTypeInstanceOrThrow(ref ppSig, out refersToNoPiaLocalType);
                     break;
 
+                case SignatureTypeCode.FunctionPointer:
+                    var signatureHeader = ppSig.ReadSignatureHeader();
+                    var parameters = DecodeSignatureParametersOrThrow(ref ppSig, signatureHeader, typeParameterCount: out int typeParamCount, shouldProcessAllBytes: false);
+                    Debug.Assert(typeParamCount == 0);
+                    Cci.CallingConvention callingConvention = signatureHeader.CallingConvention switch
+                    {
+                        SignatureCallingConvention.Default => Cci.CallingConvention.Default,
+                        SignatureCallingConvention.CDecl => Cci.CallingConvention.CDecl,
+                        SignatureCallingConvention.FastCall => Cci.CallingConvention.FastCall,
+                        SignatureCallingConvention.StdCall => Cci.CallingConvention.Standard,
+                        SignatureCallingConvention.ThisCall => Cci.CallingConvention.ThisCall,
+                        SignatureCallingConvention.VarArgs => Cci.CallingConvention.ExtraArguments,
```

Comment:
>Cci.CallingConvention.ExtraArguments [](start = 62, length = 36)

Same comment

Reply:
This one should probably just throw, I don't think we're actually planning on supporting varags function pointers in C#.

---
In reply to: [374388711](https://github.com/dotnet/roslyn/pull/41340#discussion_r374388711) [](ancestors = 374388711)

Reply:
> This one should probably just throw, I don't think we're actually planning on supporting varags function pointers in C#.

What we should be able to import/emit and what you can define in source code is not necessary the same thing. I think that up until the language is getting involved and symbol model is able to represent the metadata, we shouldn't fail

---
In reply to: [374392051](https://github.com/dotnet/roslyn/pull/41340#discussion_r374392051) [](ancestors = 374392051,374388711)